include/global.h

```diff
@@ -485,7 +485,8 @@ struct SaveBlock2
              u16 optionsBattleSceneOff:1; // whether battle animations are disabled
              u16 regionMapZoom:1; // whether the map is zoomed in
     /*0x18*/ struct Pokedex pokedex;
-    /*0x90*/ u8 filler_90[0x8];
+    /*0x90*/ u16 lastUsedBall;
```

Comment:
I tested behavior in vanilla SwSh and it doesn't keep the last used ball when saving, instead reverting to the first ball in the pocket. I feel that modifying saveblock is out of scope for BE, so only setting up EWRAM data should be enough.